templates/common/infra/bicep/core/host/container-registry.bicep

```diff
@@ -32,5 +35,31 @@ resource containerRegistry 'Microsoft.ContainerRegistry/registries@2022-02-01-pr
   }
 }
 
+resource diagnostics 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (!empty(workspaceId)) {
```

Comment:
I think this needs to be in own module, I'm trying not to embed any native resources in our modules, rather have a module that has smart defualts.

Reply:
I attempted to actually do this but ran into issue with the `scope` that is required here.  The scope needs to be available at provision time and cannot be set by a parameter.  Curious if you have run into anything similar.

Reply:
Not able to do this at the moment due to limitation in Bicep.  Issue detailed @ https://github.com/Azure/bicep/issues/622.
We would need the ability to pass a resource scope to an extension resource or would need to support string interpolation inside a existing `resource` object.